cscs-checks/prgenv/affinity_check.py

```diff
@@ -0,0 +1,155 @@
+        super().__init__(variant)
+        self.descr = 'Checking the cpu affinity for OMP threads.'
+        self.cases = {
+            'omp_bind_threads': {
+                'ref_gpu': 'gpu_omp_bind_threads.txt',
+                'ref_mc': 'mc_omp_bind_threads.txt',
+                'num_cpus_per_task': 24,
+                'ntasks-per-core': 2,
+                'multithreading': None,
+                'OMP_PLACES': 'threads',
+            },
+            'omp_bind_threads_nomultithread': {
+                'ref_gpu': 'gpu_omp_bind_threads_nomultithread.txt',
+                'ref_mc': 'mc_omp_bind_threads_nomultithread.txt',
+                'num_cpus_per_task': 12,
+                'ntasks-per-core': None,
+                # with --hint = nomultithread not explicetly expecified,
+                # this case results on using only half of the cores.
+                'multithreading': False,
+                'OMP_PLACES': 'threads',
+            },
+            'omp_bind_cores': {
+                'ref_gpu': 'gpu_omp_bind_cores.txt',
+                'ref_mc': 'mc_omp_bind_cores.txt',
+                'num_cpus_per_task': 12,
+                'ntasks-per-core': 1,
+                'multithreading': None,
+                'OMP_PLACES': 'cores',
+            },
+        }
+        self.variant = variant
+
+    def setup(self, partition, environ, **job_opts):
+        if partition.name == 'gpu':
+            self.num_cpus_per_task = (
+                self.cases[self.variant]['num_cpus_per_task'])
+        else:
+            self.num_cpus_per_task = (
+                self.cases[self.variant]['num_cpus_per_task'] * 3)
+
+        self.num_tasks = 1
+        self.variables  = {
+            'OMP_NUM_THREADS': str(self.num_cpus_per_task),
+            'OMP_PLACES': self.cases[self.variant]['OMP_PLACES']
+            # OMP_PROC_BIND is set to TRUE if OMP_PLACES is defined
+            # OMP_PROC_BIND values MASTER, CLOSE and SPREAD give the same
+            # result as OMP_PROC_BIND=TRUE
+        }
+        super().setup(partition, environ, **job_opts)
+        if self.cases[self.variant]['ntasks-per-core']:
+            self.job.options += ['--ntasks-per-core=%s' %
+                                 self.cases[self.variant]['ntasks-per-core']]
+
+
+@rfm.parameterized_test(['alternate_socket_filling'],
+                        ['consecutive_socket_filling'],
+                        ['single_task_per_socket_omp'])
+class SocketDistributionTest(AffinityTestBase):
+    def __init__(self, variant):
+        super().__init__(variant)
+        self.descr = 'Checking distribution of ranks and threads over sockets.'
+        self.valid_systems = ['daint:mc', 'dom:mc']
+        self.cases = {
+            'alternate_socket_filling': {
+                'ref_mc': 'alternate_socket_filling.txt',
+                'num_tasks': 36,
+                'num_cpus_per_task': 1,
+                'num_tasks_per_socket': 18,
+                'multithreading': False,
```

Comment:
There is no `'multithreading': True` or `'multithreading': None` test cases for `SocketDistributionTest`. Why?

Reply:
@victorusu We are not checking all different combination of flags here. This would be impractical. The purpose of the test is to use the necessary flags and environment variables in order to achieve specific and meaningful rank and thread placements.